### PR TITLE
Partially fix EYM nets: split loss working, Ustat still broken

### PR DESCRIPTION
**Pull request by g-turri**
I have added the Neural Conditional Probability (NCP) loss as described in the [Neural Conditional Probability for Inference](https://arxiv.org/abs/2407.01171) paper. Additionally, I have fixed the existing loss function based on covariances. The `"Ustat"` loss function is still not working.

### CODE REVIEW

File: kooplearn/nn/functional.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
+import torch  # noqa: E402
+from typing import Optional  # noqa: E402
 from kooplearn._src.check_deps import check_torch_deps
-
 check_torch_deps()
-from typing import Optional  # noqa: E402
-
-import torch  # noqa: E402
```
Comment (pietronvll) [inline kooplearn/nn/functional.py:6]: Importing `torch` should be done only _after_ `check_torch_deps()` has been called

File: kooplearn/nn/functional.py [hunk 2]
```diff
@@ -220,36 +236,38 @@ def eym_score_split(
...
         )
-        return score - metric_deformation * loss_on
+        return -1 * (score + metric_deformation * loss_on)
     else:
-        return score
-
+        return -1 * score
+    
 
 def eym_score(
```
Comment (pietronvll) [inline kooplearn/nn/functional.py:253]: Are there any chances that the `Ustat` mode will ever work? If not, just remove it, and every reference to it in the code.

File: kooplearn/nn/functional.py [hunk 3]
```diff
@@ -220,36 +236,38 @@ def eym_score_split(
...
+    available_modes = ["split", "Ustat"]
     if mode not in available_modes:
         raise ValueError(f"Unknown mode {mode}. Available modes are {available_modes}")
     if mode == "split":
-        return eym_score_split(X, Y, metric_deformation=metric_deformation)
-    else:
-        return eym_score_Ustat(
-            X, Y, metric_deformation=metric_deformation, center=center
-        )
+        if cov:
```
Comment (pietronvll) [inline kooplearn/nn/functional.py:265]: What's the difference between `eym_score_split_cov` and `eym_score_split`? Have you tested both, and both are working?

Does it makes sense to keep them both? If not just keep the fastest of the two.

**Review by pietronvll — CHANGES_REQUESTED** [review]

**Review by g-turri — COMMENTED** [review]
> Are there any chances that the `Ustat` mode will ever work? If not, just remove it, and every reference to it in the code.
  
I have removed `Ustat` mode and any reference to it.


> What's the difference between eym_score_split_cov and eym_score_split? Have you tested both, and both are working?

I only kept `eym_score_split_cov` since the convergence seems faster and smoother.

File: kooplearn/nn/functional.py [hunk 4]
```diff
@@ -146,70 +148,42 @@ def deepprojection_score(
...
+
+    cov_U1 = covariance(U1, center=center_covariances)
+    cov_U2 = covariance(U2, center=center_covariances)
+    cov_V1 = covariance(V1, center=center_covariances)
+    cov_V2 = covariance(V2, center=center_covariances)
+
+    cov_U1V1 = cross_covariance(U1, V1, rowvar=False, center=center_covariances)
+    cov_U2V2 = cross_covariance(U2, V2, rowvar=False, center=center_covariances)
+
+    score = 0.5 * (torch.sum(cov_U1*cov_V2) + torch.sum(cov_U2*cov_V1)) - torch.trace(cov_U1V1) - torch.trace(cov_U2V2)
```
Comment (pietronvll) [inline kooplearn/nn/functional.py:179]: Just nit-picking, but the name of this variable should be "loss" (something to be minimized) and not "score" (something to be mazimized). Am I right? 

The name of the function is eym_score, so it returns something that it should be maximized.

Comment (g-turri) [reply kooplearn/nn/functional.py]: You are right and I saw your 'Updated docs' commit. Do you prefer to keep the consistency with `vamp_score` and `deepprojection_score`, as you've done, or should we consider renaming `eym_score` function (and variable therein) to `loss` for clarity? 'Loss' might be more intuitive, especially since the NCP paper also refers to it in those terms.

Comment (pietronvll) [reply kooplearn/nn/functional.py]: Just rename the variable within to `loss` and we'll be consistent with the rest of the package

**Review by pietronvll — APPROVED** [review]

**Comment by pietronvll** [comment]
Good to go, merging into `dev`
